### PR TITLE
Send productClick events to PixelManager

### PR DESCRIPTION
**Pull request by italo-batista**
#### What is the purpose of this pull request?

Sending productClick event to Pixel Manager with the purpose of usage in google-analytics.

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [X] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

**Workspace**: https://persepolis--storecomponents.myvtex.com

### CODE REVIEW

File: react/components/GalleryItem.js [hunk 1]
```diff
@@ -9,8 +10,8 @@ import { PropTypes } from 'prop-types'
 /**
  * Normalizes the item received in the props to adapt to the extension point prop.
  */
-const GalleryItem = ({ item, displayMode, summary }) => {
-  const normalizeProductSummary = product => {
+class GalleryItem extends Component {
```
Comment (lucasecdb) [inline react/components/GalleryItem.js]: why did you move it to a class component?

File: react/components/GalleryItem.js [hunk 2]
```diff
@@ -49,17 +50,29 @@ const GalleryItem = ({ item, displayMode, summary }) => {
...
+        product={this.normalizeProductSummary(item)}
+        displayMode={displayMode}
+        actionOnClick={() => this.sendProductClickEvent(item)}
+      />
+    )
+  }
 }
 
-GalleryItem.propTypes = {
+export const propTypes = {
```
Comment (lucasecdb) [inline react/components/GalleryItem.js]: you removed the prop types from the component.

and why do you need to export this?

File: react/components/GalleryItem.js (hunk 1 above)
Comment (italo-batista) [reply react/components/GalleryItem.js]: Because the Pixel "interface" should have access to a component's props to push notifications.

File: react/components/GalleryItem.js (hunk 2 above)
Comment (italo-batista) [reply react/components/GalleryItem.js]: I removed because Pixel overwritten it anyway, but I exported to make it still available if other components need it.

File: react/components/GalleryItem.js (hunk 1 above)
Comment (italo-batista) [reply react/components/GalleryItem.js]: I will follow the suggestion of @jgfidelis and use usePixel. Didn't know its existance. :)

File: react/components/GalleryItem.js (hunk 2 above)
Comment (lucasecdb) [reply react/components/GalleryItem.js]: no, the pixel won't override the component's propTypes, it creates a _new_ component that wraps this one, the propTypes are not lost in the way

File: react/components/GalleryItem.js (hunk 1 above)
Comment (lucasecdb) [reply react/components/GalleryItem.js]: by the way, you can still use high order components in function components, its the same as you would use in a class. there was no need to convert it into a class even if the `usePixel` didn't exist

Comment (lucasecdb) [reply react/components/GalleryItem.js]: check [this guide](https://reactjs.org/docs/higher-order-components.html) if you want to learn more

File: react/components/GalleryItem.js [hunk 3]
```diff
@@ -49,12 +50,19 @@ const GalleryItem = ({ item, displayMode, summary }) => {
     return normalizedProduct
   }
 
+  const { push } = usePixel()
+
+  const sendProductClickEvent = product => {
+    push({ event: 'productClick', product: product })
```
Comment (jgfidelis) [inline react/components/GalleryItem.js]:
```suggestion
    push({ event: 'productClick', product })
```
